tests/solidity/init-node.sh

```diff
@@ -0,0 +1,168 @@
+evmosd add-genesis-account "$(evmosd keys show "$VAL_KEY" -a --keyring-backend "$KEYRING")" 100000000000000000000000000aevmos --keyring-backend "$KEYRING"
+evmosd add-genesis-account "$(evmosd keys show "$USER1_KEY" -a --keyring-backend "$KEYRING")" 1000000000000000000000aevmos --keyring-backend "$KEYRING"
+evmosd add-genesis-account "$(evmosd keys show "$USER2_KEY" -a --keyring-backend "$KEYRING")" 1000000000000000000000aevmos --keyring-backend "$KEYRING"
+evmosd add-genesis-account "$(evmosd keys show "$USER3_KEY" -a --keyring-backend "$KEYRING")" 1000000000000000000000aevmos --keyring-backend "$KEYRING"
+evmosd add-genesis-account "$(evmosd keys show "$USER4_KEY" -a --keyring-backend "$KEYRING")" 1000000000000000000000aevmos --keyring-backend "$KEYRING"
+
+# Update total supply with claim values
+# Bc is required to add this big numbers
+# total_supply=$(bc <<< "$amount_to_claim+$validators_supply")
+total_supply=100004000000000000000010000
+jq -r --arg total_supply "$total_supply" '.app_state.bank.supply[0].amount=$total_supply' "$GENESIS" >"$TMP_GENESIS" && mv "$TMP_GENESIS" "$GENESIS"
+
+# set custom pruning settings
+if [ "$PRUNING" = "custom" ]; then
+  sed -i.bak 's/pruning = "default"/pruning = "custom"/g' "$APP_TOML"
+  sed -i.bak 's/pruning-keep-recent = "0"/pruning-keep-recent = "2"/g' "$APP_TOML"
+  sed -i.bak 's/pruning-interval = "0"/pruning-interval = "10"/g' "$APP_TOML"
+fi
+
+# make sure the localhost IP is 0.0.0.0
+sed -i.bak 's/localhost/0.0.0.0/g' "$CONFIG_TOML"
+sed -i.bak 's/127.0.0.1/0.0.0.0/g' "$APP_TOML"
+
+# use timeout_commit 1s to make test faster
+sed -i.bak 's/timeout_commit = "3s"/timeout_commit = "1s"/g' "$CONFIG_TOML"
+
+# Sign genesis transaction
+evmosd gentx "$VAL_KEY" 1000000000000000000000aevmos --gas-prices ${BASEFEE}aevmos --keyring-backend "$KEYRING" --chain-id "$CHAINID"
+## In case you want to create multiple validators at genesis
+## 1. Back to `evmosd keys add` step, init more keys
+## 2. Back to `evmosd add-genesis-account` step, add balance for those
+## 3. Clone this ~/.evmosd home directory into some others, let's say `~/.clonedEvmosd`
+## 4. Run `gentx` in each of those folders
+## 5. Copy the `gentx-*` folders under `~/.clonedEvmosd/config/gentx/` folders into the original `~/.evmosd/config/gentx`
+
+# Enable the APIs for the tests to be successful
+sed -i.bak 's/enable = false/enable = true/g' "$APP_TOML"
+
+# Don't enable memiavl by default
+grep -q -F '[memiavl]' "$APP_TOML" && sed -i.bak '/\[memiavl\]/,/^\[/ s/enable = true/enable = false/' "$APP_TOML"
+
+# Collect genesis tx
+evmosd collect-gentxs
+
+# Run this to ensure everything worked and that the genesis file is setup correctly
+evmosd validate-genesis
+
+# Start the node
+evmosd start "$TRACE" \
+  --log_level $LOGLEVEL \
+  --minimum-gas-prices=0.0001aevmos \
+  --json-rpc.api eth,txpool,personal,net,debug,web3 \
+  --json-rpc.allow-insecure-unlock true \
+  --chain-id "$CHAINID"
```

Comment:
not sure why this whole file was added - as far as I can tell this functionality is not made use of in tests? If I'm mistaken, please point me to where the `--json-rpc.allow-insecure-unlock` has an effect in tests

Reply:
Previously, It used init-node.sh under e2e, but because e2e has TestUpgrade, it will use the old evmos binary file. If add “--json-rpc.allow-insecure-unlock true” directly to e2e, an error will occur.